kernel/aster-nix/src/net/socket/unix/stream/listener.rs

```diff
@@ -12,57 +14,55 @@ use crate::{
 };
 
 pub(super) struct Listener {
-    addr: UnixSocketAddrBound,
+    backlog: Arc<Backlog>,
 }
 
 impl Listener {
-    pub(super) fn new(addr: UnixSocketAddrBound, backlog: usize) -> Result<Self> {
-        BACKLOG_TABLE.add_backlog(&addr, backlog)?;
-        Ok(Self { addr })
+    pub(super) fn new(addr: UnixSocketAddrBound, backlog: usize) -> Self {
+        let backlog = BACKLOG_TABLE.add_backlog(addr, backlog).unwrap();
```

Comment:
if `add_backlog` returns `None`, should `Errno::ADDRINUSE` be returned here?

Reply:
Without race conditions, I don't think it's possible to get a `None` here.

If we bind two sockets to the same address, the second `bind()` will result in an `EADDRINUSE`.

If we do two simultaneous `listen()` calls on a bound socket, the second `listen()` will see the `Listen` state directly, so it won't try to add itself to the backlog table.

So I use `unwrap()` here. I think it's an internal error if we get `None`.